src/eduid_webapp/group_management/helpers.py

```diff
@@ -18,10 +29,142 @@ class GroupManagementMsg(TranslatableMsg):
+        raise EduIDDBError('Failed to save group')
+
+    current_app.logger.info(f'Added user as {invite.role} to group with scim_id: {invite.group_scim_id}')
+    return None
+
+
+def remove_user_from_group(scim_user: ScimApiUser, scim_group: ScimApiGroup, role: str) -> None:
+    if role == GroupRole.OWNER.value:
+        if is_owner(scim_user, scim_group.scim_id):
+            scim_group.graph.owners = [
+                owner for owner in scim_group.graph.owners if owner.identifier != str(scim_user.scim_id)
+            ]
+
+    elif role == GroupRole.MEMBER.value:
+        if is_member(scim_user, scim_group.scim_id):
+            scim_group.graph.members = [
+                member for member in scim_group.graph.members if member.identifier != str(scim_user.scim_id)
+            ]
+    else:
+        raise NotImplementedError(f'Unknown role: {role}')
+
+    if not current_app.scimapi_groupdb.save(scim_group):
+        raise EduIDDBError(f'Failed to save group with scim_id {scim_group.scim_id}')
+
+    current_app.logger.info(
+        f'Removed user as with scim_id {scim_user.scim_id} as {role} from group with scim_id {scim_group.scim_id}'
+    )
+    return None
+
+
+def get_outgoing_invites(groups: List[ScimApiGroup]) -> List[Dict[str, Any]]:
+    """
+    Return all outgoing invites to groups that the user is owner of.
+    """
+    invites = []
+    for group in groups:
+        try:
+            states = current_app.invite_state_db.get_states_by_group_scim_id(str(group.scim_id))
+        except DocumentDoesNotExist:
+            continue
+        group_invite = {'identifier': group.scim_id, 'owner_invites': [], 'member_invites': []}
+        for state in states:
+            if state.role == GroupRole.OWNER.value:
+                group_invite['owner_invites'].append({'email_address': state.email_address})
+            if state.role == GroupRole.MEMBER.value:
+                group_invite['member_invites'].append({'email_address': state.email_address})
+        invites.append(group_invite)
+    current_app.logger.info(f'outgoing invites: {invites}')
+    return invites
+
+
+def get_incoming_invites(user: User) -> List[Dict[str, Any]]:
+    """
+    Return all incoming invites to groups for the user
+    """
+    invites = []
+    emails = [item.email for item in user.mail_addresses.to_list() if item.is_verified]
+    states = current_app.invite_state_db.get_states_by_email_addresses(emails, raise_on_missing=False)
+    for state in states:
+        group = current_app.scimapi_groupdb.get_group_by_scim_id(state.group_scim_id)
+        if group is None:
+            current_app.invite_state_db.remove_state(state)
+            current_app.logger.info(f'Removed invite to non existant group: {state}')
+            continue
+
+        owners = [{'identifier': owner.identifier, 'display_name': owner.display_name} for owner in group.graph.owners]
+        invites.append(
```

Comment:
maybe this kind of complex return value warrants a dataclass? and then probably the function above too.